jax/api_util.py

```diff
@@ -70,3 +71,23 @@ def flatten_fun_nokwargs2(in_tree, *args_flat):
   ans_flat, ans_tree = tree_flatten(ans)
   aux_flat, aux_tree = tree_flatten(aux)
   yield (ans_flat, aux_flat), (ans_tree, aux_tree)
+
+def flatten_axes(treedef, axis_tree):
```

Comment:
For functions like this I think that it is worth starting to add unit tests. We do most of our testing with integration tests and it is hard to know what kind of coverage we have. Also, unit tests can be helpful to understand in detail the behavior of this function.